src/types.rs

```diff
@@ -77,7 +77,8 @@ macro_rules! new_type {
         }
     ) => {
         $(#[$attr])*
-        #[derive(Clone, Debug, PartialEq)]
+        #[derive(Clone, Debug, PartialEq, Eq, Hash, ::serde::Serialize, ::serde::Deserialize)]
```

Comment:
I think there was briefly a time when not all of the new types had these traits derived, but that no longer seems to be the case, so 👍